client/src/app/services/auth.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from "@angular/common/http";
+import { User } from "../interfaces/user";
+import { environment } from "../../environments/environment";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  private url = environment.appUrl;
+  private token: string | boolean = false;
+  private user: any;
+
+  private headers: HttpHeaders = new HttpHeaders();
+  private options: any;
+
+  constructor(private http: HttpClient) {}
+
+  ngOnInit() {
+    this.setHeaders()
+  }
+
+  login(data: any): any {
+    return this.http.post(this.url + '/login', data, this.options);
+  }
+
+  register(data: any):any {
+    return this.http.post(this.url + '/register', data, this.options);
+  }
+
+  private setHeaders() {
+    this.headers = new HttpHeaders({
+      'Content-Type': 'application/json',
+      'Authorization': 'Bearer ' + this.token
+    });
+    this.options = { headers: this.headers };
+  }
```

Comment:
This can be implemented with [interceptors](https://medium.com/@jaydeepvpatil225/http-interceptors-in-angular-6e9891ae0538)